src/libcore/iter.rs

```diff
@@ -2655,6 +2656,7 @@ pub fn range_step<A: Int>(start: A, stop: A, step: A) -> RangeStep<A> {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
+#[allow(deprecated)]
```

Comment:
My only worry is that this sort of usage of `Int` is not warned about in the unstable lint, so we could alter this functionality of code using 100% stable code and we could still break them. This is pretty minor breakage though so it's not necessarily that bad.